Snapshots from dcp25 & dcp27 contain `||` in metadata

Errors experienced during reindex of `dcp27` catalog

```
[ERROR] RequirementError: ("'||' is disallowed in metadata", TDRBundleFQID(uuid='99abef08-9516-4831-a5c0-685ec357323b', version='2022-08-10T09:53:22.830000Z', source=SourceRef(id='bab5eaa4-c480-49b4-a6dc-4abd02f2f7ec', spec=TDRSourceSpec(prefix=Prefix(common='', partition=1), project='datarepo-96f28c63', name='hca_prod_615158205bb845d08d12f0850222ecf0__20221007_dcp2_20230314_dcp25', is_snapshot=True))))
Traceback (most recent call last):
  File "/var/task/azul/chalice.py", line 142, in patched_event_source_handler
    return old_handler(self_, event, context)
  File "/var/task/chalice/app.py", line 1752, in __call__
    return self.handler(event_obj)
  File "/var/task/chalice/app.py", line 1708, in __call__
    return self.handler(request, self.next_handler)
  File "/var/task/azul/chalice.py", line 193, in _lambda_context_middleware
    return get_response(event)
  File "/var/task/app.py", line 188, in contribute_retry
    app.index_controller.contribute(event, retry=True)
  File "/var/task/azul/indexer/index_controller.py", line 179, in contribute
    contributions = self.transform(catalog, notification, delete)
  File "/var/task/azul/indexer/index_controller.py", line 214, in transform
    results = service.transform(catalog, bundle, partition, delete=delete)
  File "/var/task/azul/indexer/index_service.py", line 263, in transform
    bundle.reject_joiner()
  File "/var/task/azul/indexer/__init__.py", line 457, in reject_joiner
    self._reject_joiner(self.metadata_files)
  File "/var/task/azul/indexer/__init__.py", line 443, in _reject_joiner
    self._reject_joiner(v)
  File "/var/task/azul/indexer/__init__.py", line 443, in _reject_joiner
    self._reject_joiner(v)
  File "/var/task/azul/indexer/__init__.py", line 443, in _reject_joiner
    self._reject_joiner(v)
  File "/var/task/azul/indexer/__init__.py", line 450, in _reject_joiner
    raise RequirementError(msg, self.fqid)
```

**CloudWatch Logs Insights**  
region: us-east-1  
log-group-names: /aws/lambda/azul-indexer-prod, /aws/lambda/azul-indexer-prod-aggregate, /aws/lambda/azul-indexer-prod-aggregate_retry, /aws/lambda/azul-indexer-prod-contribute, /aws/lambda/azul-indexer-prod-contribute_retry  
start-time: -86400s  
end-time: 0s  
query-string:
```
fields @timestamp, @message
| filter @message like 'disallowed in metadata'
| parse "name='*'" as snapshot
| stats count(*) by snapshot

```
---
| snapshot | count(*) |
| --- | --- |
| hca_prod_615158205bb845d08d12f0850222ecf0__20221007_dcp2_20230314_dcp25 | 66998 |
| hca_prod_5f607e50ba224598b1e9f3d9d7a35dcc__20230201_dcp2_20230501_dcp27 | 44 |
| hca_prod_e6773550c1a6494986431a3154cf2670__20221208_dcp2_20230314_dcp25 | 40 |
---

Timeline:
```
05/10/2023  dcp27 added: 5f607e50ba224598b1e9f3d9d7a35dcc added
05/04/2023  require no ||, no reindex done
04/20/2023  dcp26 added
03/24/2023  dcp25 added: 615158205bb845d08d12f0850222ecf0 updated, e6773550c1a6494986431a3154cf2670 updated
12/12/2022  dcp23 added: e6773550c1a6494986431a3154cf2670 added
10/12/2022  dcp21 added: 615158205bb845d08d12f0850222ecf0 added
```

----

- [ ] Security design review completed; the Resolution of this issue does **not** …
  - [ ] … affect authentication; for example:
    - OAuth 2.0 with the application (API or Swagger UI)
    - Authentication of developers with Google Cloud APIs
    - Authentication of developers with AWS APIs
    - Authentication with a GitLab instance in the system
    - Password and 2FA authentication with GitHub
    - API access token authentication with GitHub
    - Authentication with 
  - [ ] … affect the permissions of internal users like access to
    - Cloud resources on AWS and GCP
    - GitLab repositories, projects and groups, administration
    - an EC2 instance via SSH
    - GitHub issues, pull requests, commits, commit statuses, wikis, repositories, organizations
  - [ ] … affect the permissions of external users like access to
    - TDR snapshots
  - [ ] … affect permissions of service or bot accounts
    - Cloud resources on AWS and GCP
  - [ ] … affect audit logging in the system, like
    - adding, removing or changing a log message that represents an auditable event
    - changing the routing of log messages through the system
  - [ ] … affect monitoring of the system
  - [ ] … introduce a new software dependency like
    - Python packages on PYPI
    - Command-line utilities
    - Docker images
    - Terraform providers
  - [ ] … add an interface that exposes sensitive or confidential data at the security boundary
  - [ ] … affect the encryption of data at rest
  - [ ] … require persistence of sensitive or confidential data that might require encryption at rest
  - [ ] … require unencrypted transmission of data within the security boundary
  - [ ] … affect the network security layer; for example by 
    - modifying, adding or removing firewall rules
    - modifying, adding or removing security groups
    - changing or adding a port a service, proxy or load balancer listens on
- [ ] Documentation on any unchecked boxes is provided in comments below

--- a/deployments/prod/environment.py
+++ b/deployments/prod/environment.py
@@ -780,15 +780,17 @@ dcp25_sources = mkdict(dcp24_sources, 333, mkdelta([
     mksrc('datarepo-289f5713', 'hca_prod_fde199d2a8414ed1aa65b9e0af8969b1__20220330_dcp2_20230314_dcp25', 186),
 ]))
 
-dcp26_sources = mkdict(dcp25_sources, 336, mkdelta([
+dcp26_sources = mkdict(dcp25_sources, 334, mkdelta([
     mksrc('datarepo-37928b8f', 'hca_prod_2d4d89f2ebeb467cae60a3efc5e8d4ba__20230206_dcp2_20230331_dcp26', 2),
+    mksrc('datarepo-96f28c63', 'hca_prod_615158205bb845d08d12f0850222ecf0__20221007_dcp2_20230314_dcp25', 5174, pop),
     mksrc('datarepo-0d0c66d4', 'hca_prod_c281ab637b7d4bdfb7619b1baaa18f82__20230331_dcp2_20230331_dcp26', 2),
     mksrc('datarepo-b3b1e92f', 'hca_prod_c5b475f276b34a8e8465f3b69828fec3__20230331_dcp2_20230331_dcp26', 11),
     mksrc('datarepo-baa51c1d', 'hca_prod_cd9d6360ce38432197dff13c79e3cb84__20230206_dcp2_20230331_dcp26', 2),
+    mksrc('datarepo-d0d87d10', 'hca_prod_e6773550c1a6494986431a3154cf2670__20221208_dcp2_20230314_dcp25', 56, pop),
     mksrc('datarepo-636717a5', 'hca_prod_f3825dfe990a431fb9719c26d39840db__20230331_dcp2_20230331_dcp26', 31),
 ]))
 
-dcp27_sources = mkdict(dcp26_sources, 353, mkdelta([
+dcp27_sources = mkdict(dcp26_sources, 350, mkdelta([
     mksrc('datarepo-75b50ae7', 'hca_prod_04e4292cf62f4098ae9bfd69ae002a90__20230427_dcp2_20230501_dcp27', 310),
     mksrc('datarepo-d2f36342', 'hca_prod_0751843070314bdfa3ce1bf0917a1923__20221208_dcp2_20230501_dcp27', 7),
     mksrc('datarepo-26882f0f', 'hca_prod_07d5987e7f9e4f34b0fba185a35504f5__20230427_dcp2_20230501_dcp27', 36),
@@ -801,7 +803,6 @@ dcp27_sources = mkdict(dcp26_sources, 353, mkdelta([
     mksrc('datarepo-a498b8c7', 'hca_prod_50154d1e230844bf960810c7afaa560b__20230427_dcp2_20230501_dcp27', 322),
     mksrc('datarepo-f800a6a6', 'hca_prod_566d00b0e1f84b929cbd57de9fad0050__20230427_dcp2_20230501_dcp27', 24),
     mksrc('datarepo-bf159952', 'hca_prod_5f44a860d96e4a99b67e24e1b8ccfd26__20230427_dcp2_20230501_dcp27', 3),
-    mksrc('datarepo-a9c85bb3', 'hca_prod_5f607e50ba224598b1e9f3d9d7a35dcc__20230201_dcp2_20230501_dcp27', 10),
     mksrc('datarepo-319b1c54', 'hca_prod_6e1771950ac0468b99a287de96dc9db4__20230503_dcp2_20230503_dcp27', 412),
     mksrc('datarepo-7eec55b7', 'hca_prod_77c13c40a5984036807fbe09209ec2dd__20230427_dcp2_20230501_dcp27', 40),
     mksrc('datarepo-daba93c4', 'hca_prod_7c5990297a3c4b5c8e79e72c9a9a65fe__20230427_dcp2_20230501_dcp27', 18),

